### PR TITLE
8257232: CompileThresholdScaling fails to work on 32-bit platforms

### PR DESCRIPTION
**Pull request by DamonFool**
Hi all,

CompileThresholdScaling is incorrect on 32-bit platforms.

If you run the following command on Linux-32:
```
java -XX:CompileThresholdScaling=0.75 -version
```
It gets the following unexpected warnings:
```
intx Tier0InvokeNotifyFreqLog=32 is outside the allowed range [ 0 ... 30 ]
intx Tier0BackedgeNotifyFreqLog=32 is outside the allowed range [ 0 ... 30 ]
intx Tier2InvokeNotifyFreqLog=32 is outside the allowed range [ 0 ... 30 ]
intx Tier2BackedgeNotifyFreqLog=32 is outside the allowed range [ 0 ... 30 ]
intx Tier3InvokeNotifyFreqLog=32 is outside the allowed range [ 0 ... 30 ]
intx Tier3BackedgeNotifyFreqLog=32 is outside the allowed range [ 0 ... 30 ]
intx Tier23InlineeNotifyFreqLog=32 is outside the allowed range [ 0 ... 30 ]
```

The failure is that nth_bit(max_freq_bits) [1] = nth_bit(32) [2] = 0 on 32-bit platforms.
So the scaling logic is wrong.
It would be better to fix it.

Thanks.
Best regards,
Jie

[1] https://github.com/openjdk/jdk/blob/master/src/hotspot/share/compiler/compilerDefinitions.cpp#L125
[2] https://github.com/openjdk/jdk/blob/master/src/hotspot/share/utilities/globalDefinitions.hpp#L973

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8257232](https://bugs.openjdk.java.net/browse/JDK-8257232): CompileThresholdScaling fails to work on 32-bit platforms

### Reviewers
 * [Vladimir Kozlov](https://openjdk.java.net/census#kvn) (@vnkozlov - **Reviewer**)
 * [Claes Redestad](https://openjdk.java.net/census#redestad) (@cl4es - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/1499/head:pull/1499`
`$ git checkout pull/1499`

### CODE REVIEW

**Review by vnkozlov — APPROVED** [review]
Good.

**Comment by DamonFool** [comment]
> Good.

Thanks @vnkozlov for your review.

May I get a second review for this change?
Thanks.

**Review by cl4es — APPROVED** [review]
Looks good to me.

Just an idea for a future RFE (one which I might pick up and work on myself): If the multiply-loop-based `log2_intptr` is a performance issue here, it could probably be optimized by building a variant based on `count_leading_zeros`. That would require lifting it out from `globalDefinitions.hpp`, so `powerOfTwo.hpp` might be a good place.

File: src/hotspot/share/compiler/compilerDefinitions.cpp
Comment (DamonFool) [reply src/hotspot/share/compiler/compilerDefinitions.cpp:127]: > Could be simplified to `return MIN2(log2_intptr(scaled_freq), max_freq_bits);`

OK.
Will use this kind of code style next time.
Thanks.

**Pull request closed without merging**
